core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -258,9 +260,55 @@ private Component renderTitle(final Config config, final MapInfo map) {
       return header.colorIfAbsent(NamedTextColor.AQUA);
     }
 
-    final Component game = map.getGamemode();
-    if (game != null) {
-      return game.colorIfAbsent(NamedTextColor.AQUA);
+    final Component gamemode = map.getGamemode();
+    if (gamemode != null) {
+      return gamemode.colorIfAbsent(NamedTextColor.AQUA);
+    }
+
+    final Collection<Gamemode> gamemodes = map.getGamemodes();
+    if (!gamemodes.isEmpty()) {
+      Component gamemodeTitle = null;
+      List<Component> gamemodeComponents = new ArrayList<>();
+      for (Gamemode value : gamemodes) {
+        Component gmComponent;
+        // Make this smarter using MAX_LENGTH
+        if (gamemodes.size() <= 1) {
+          gmComponent = Component.translatable("gamemode." + value.getId() + ".name");
+        } else {
+          gmComponent = Component.translatable("gamemode." + value.getId() + ".acronym");
+        }
+        gamemodeComponents.add(gmComponent);
+      }
```

Comment:
```suggestion
      String suffix = gamemodes.size() <= 1 ? ".name" : ".acronym";
      List<Component> gmComponents = gamemodes.stream()
              .map(gm  -> translatable("gamemode." + gm.getId() + suffix))
              .collect(Collectors.toList());      
      Component gamemodeTitle = null;
```

You can simplify all of that into this
There's no need for an if-check on each iteration to decide what suffix to use (moved it out of the loop), and there's no need to have a list that resizes on each iteration